loader/loader.c

```diff
@@ -5127,6 +5127,149 @@ bool loaderGetLayerInterfaceVersion(PFN_vkNegotiateLoaderLayerInterfaceVersion f
     return true;
 }
 
+VKAPI_ATTR VkResult VKAPI_CALL vkLayerCreateDevice(VkInstance instance, VkPhysicalDevice physicalDevice, const VkDeviceCreateInfo *pCreateInfo,
+						   const VkAllocationCallbacks *pAllocator, VkDevice *pDevice, PFN_vkGetInstanceProcAddr layerGIPA, PFN_vkGetDeviceProcAddr *nextGDPA) {
```

Comment:
I have a couple of thoughts on the prototype for this function:
- I'd like it better if this function's name didn't look the same as a Vulkan API call, since it isn't. I know we've done that in other places, but I'd prefer not to keep doing it. I would recommend giving it a name along the lines of `LoaderInitDevice`. Feel free not to use that exact name if you have a better suggestion.
- When I looked at the `PFN_vkGetInstanceProcAddr` parameter and the way it's used to tell the loader where to stop building the device chain, I wasn't crazy about it. But the more I've thought about it the more I feel like we have to do it this way because the device chain doesn't include much other info. So I think this is fine, then.